pkg/ovs/ovn-nb-load_balancer.go

```diff
@@ -0,0 +1,259 @@
+package ovs
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/ovn-org/libovsdb/model"
+	"github.com/ovn-org/libovsdb/ovsdb"
+
+	ovsclient "github.com/kubeovn/kube-ovn/pkg/ovsdb/client"
+	"github.com/kubeovn/kube-ovn/pkg/ovsdb/ovnnb"
+)
+
+// CreateLoadBalancer create loadbalancer
+func (c *ovnClient) CreateLoadBalancer(lbName, protocol, selectFields string) error {
+	exist, err := c.LoadBalancerExists(lbName)
+	if err != nil {
+		return err
+	}
+
+	// found, ignore
+	if exist {
+		return nil
+	}
+
+	lb := &ovnnb.LoadBalancer{
+		UUID:     ovsclient.NamedUUID(),
+		Name:     lbName,
+		Protocol: &protocol,
+	}
+
+	if len(selectFields) != 0 {
+		lb.SelectionFields = []string{selectFields}
+	}
+
+	op, err := c.ovnNbClient.Create(lb)
+	if err != nil {
+		return fmt.Errorf("generate operations for creating load balancer %s: %v", lbName, err)
+	}
+
+	if err := c.Transact("lb-add", op); err != nil {
+		return fmt.Errorf("create load balancer %s: %v", lbName, err)
+	}
+
+	return nil
+}
+
+// UpdateLoadBalancer update load balancer
+func (c *ovnClient) UpdateLoadBalancer(lb *ovnnb.LoadBalancer, fields ...interface{}) error {
+	op, err := c.ovnNbClient.Where(lb).Update(lb, fields...)
+	if err != nil {
+		return fmt.Errorf("generate operations for updating load balancer %s: %v", lb.Name, err)
+	}
+
+	if err = c.Transact("lb-update", op); err != nil {
+		return fmt.Errorf("update load balancer %s: %v", lb.Name, err)
+	}
+
+	return nil
+}
+
+// LoadBalancerAddVips add vips
+func (c *ovnClient) LoadBalancerAddVips(lbName string, vips map[string]string) error {
+	if len(vips) == 0 {
+		return nil
+	}
+
+	lb, err := c.GetLoadBalancer(lbName, false)
+	if err != nil {
+		return err
+	}
+
+	if lb.Vips == nil {
+		lb.Vips = make(map[string]string)
+	}
+
+	for vip, backends := range vips {
+		lb.Vips[vip] = backends
+	}
```

Comment:
```suggestion
updatedVips := make(map[string]string, len(lb.Vips)+1)
	for vip, backends := range lb.Vips {
		updatedVips[vip] = backends
	}
	for vip, backends := range vips {
		updatedVips[vip] = backends
	}
	lb.Vips = updatedVips
```

Updating the filed directly will cause unexpected result.